server/core_tournament.go

```diff
@@ -737,22 +737,17 @@ func calculateTournamentDeadlines(startTime, endTime, duration int64, resetSched
 			tUnix = t.UTC().Unix()
 		}
 
-		schedules := resetSchedule.NextN(t, 2)
-		// Roll time back a safe amount, then scan forward looking for the current start active.
-		startActiveUnix := tUnix - ((schedules[1].UTC().Unix() - schedules[0].UTC().Unix()) * 2)
-		for {
-			s := resetSchedule.Next(time.Unix(startActiveUnix, 0).UTC()).UTC().Unix()
-			if s < tUnix {
-				startActiveUnix = s
-			} else {
-				if s == tUnix {
-					startActiveUnix = s
-				}
-				break
-			}
+		var startActiveUnix int64
+		// check if we are landing squarely on the reset schedule
+		if resetSchedule.Next(t.Add(-1*time.Second)) == t {
+			startActiveUnix = tUnix
+		} else {
+			// otherwise assume the supplied time is ahead of the reset schedule.
```

Comment:
I believe this assumption was baked into the for loop of the previous code as well.

Reply:
Is this covering the case where `startActiveTime` is in the future?

Reply:
Seeing the leaderboard code, it looks like `0` should be returned if `startActiveTime` is in the future.